### PR TITLE
Add Prolog

### PR DESCRIPTION
**Pull request by aarroyoc**
I'm not sure if Prolog counts as there's no compilation step. However, the proof is the code itself, which is interesting too. As long as the proof is OK, the code will be OK because the programming language itself is made of assertions. It's not possible to have a well written proof and then have a failing example. Also, modifications to the code that made some of the assertions of the proof fail, will make the predicate fail as a whole.

### CODE REVIEW

File: prolog/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Prolog
+
+Prolog is a logic programming language. It is based on First Order Logic and features unification and non-determinism.
+
+A Prolog program is made of predicates. Each predicate can be one or more clauses, and each clause can be a rule or a fact. For a predicate to be true, at least one of the clauses must be true. A fact is true by itself, and a rule is true iff the body of the rule is true.
+
+The program itself it's a transcription of the spec to Prolog code. Note that we should not give
```
Comment (waldyrious) [inline prolog/README.md:7]:
```suggestion
The program itself is a transcription of the spec to Prolog code. Note that we should not give
```

**Comment by hwayne** [comment]
Thank you for the submission! Unfortunately, I can't accept this submission. As written in a prior [prolog submission](https://github.com/hwayne/lets-prove-leftpad/pull/30):

> Prolog is not actually a theorem prover, it just so happens that this problem is very, very easy to express in Prolog in a simple way. If the question was "sort a list of numbers", a Python solution that was just `sorted(l)` wouldn't count for the same reasons.

That said, your solution was almost entirely identical to his, so it's interesting that the "Prolog way" for this converges so nicely. Maybe that could be a blog post of its own!

**Comment by triska** [comment]
Regarding the statement "Prolog is not actually a theorem prover": How is Prolog not actually a theorem prover? As long as we keep to the pure monotonic core of Prolog (which the program in this PR does), the answers to queries are logical consequences of the given program, and we get *only* logical consequences of the program in response to queries.

In its default execution strategy, Prolog may not be a particularly *smart* theorem prover (because it sometimes does not terminate even though a different execution strategy *would* terminate), but the argument here is that it is not a theorem prover at all?

**Comment by hwayne** [comment]
I should actually write a writeup of why it's not, which will happen eventually

**Comment by triska** [comment]
May I suggest you instead plan a writeup about why it is? Thank you a lot!

**Comment by hwayne** [comment]
We'll see where this ends up :p

**Pull request closed without merging**